include/highfive/bits/H5Node_traits.hpp

```diff
@@ -107,13 +114,21 @@ class NodeTraits {
     /// \return number of leaf objects
     std::vector<std::string> listObjectNames() const;
 
-    ///    
+    ///
     /// \brief check a dataset or group exists in the current node / group
     /// \param dataset/group name to check
     /// \return true if a dataset/group with the asssociated name exist, or
     /// false
     bool exist(const std::string& node_name) const;
 
+    /// \brief Checks if a node / path exist under the current node / group.
+    /// \param node / path name to check
+    /// \param Whether recursive checking shall be activated. When disabled
+    ///        plain H5Lexists is used, which fails if a part of the path 
+    ////       doesnt exist
+    /// \return true if the given node name / path exists
+    bool exist(const std::string& node_path, bool recursive) const;
```

Comment:
Does it make sense to only have one `exist` member method with an `recursive` parameter with default value `false`?

```cpp
bool exist(const std::string& node_path, bool recursive = false) const;
```

Besides Doxygen parameter documentation is not consistent between the 2 `exist` member methods.

Reply:
According to this page: https://support.hdfgroup.org/HDF5/doc/H5.user/Groups.html, which I assume to be latest version of the library, slashes are not allowed as part of any component name. I would make have a single function that handles recursion by default. That shouldn't have any noticeable performance impact.

Reply:
Thanks @hernando for the link, I was missing verifying that info.
Information in https://support.hdfgroup.org/HDF5/doc/RM/RM_H5L.html#Link-Exists is a bit misleading. They even made `exist("/")` at the file and root node level to return 1.
I agree that having overloads might not be very elegant. Putting together your suggestions I feel we should have a different function name for the base version to become obviously non-recursive, like `exist_node` and turn `exist` always recursive. What you think?

Reply:
Reading  bit more Groups.html it is understandable that names can be composed and have slashes, and H5Lexists will respect that. The only problem is indeed when middle components don't exist, where instead of returning false it throws an exception.
Without knowing the motivation behind that, we could simply fix that case. 
I'll make some changes.